services/apps/src/apps/service/apps/de/listings.clj

```diff
@@ -242,9 +258,10 @@
    descendents."
   [user app-group-id params]
   (let [workspace (get-optional-workspace (:username user))
-        params    (fix-sort-params params)]
-    (or (list-apps-in-virtual-group user workspace app-group-id params)
-        (list-apps-in-real-group user workspace app-group-id params))))
+        perms     (iplant-groups/load-app-permissions (:shortUsername user))
+        params    (fix-sort-params (assoc params :app-ids (set (keys perms))))]
```

Comment:
We're associating the set of app IDs that the user can view with each app or category listing request. Simply adding the IDs in the request parameters was the easiest way to pass the information to the code that is performing the query.